js/src/common/utils/mapRoutes.ts

```diff
@@ -6,20 +8,23 @@ import DefaultResolver from '../resolvers/DefaultResolver';
  * to provide each route with the current route name.
  *
  * @see https://mithril.js.org/route.html#signature
- * @param {Object} routes
- * @param {String} [basePath]
- * @return {Object}
  */
-export default function mapRoutes(routes, basePath = '') {
-  const map = {};
+export default function mapRoutes(
+  routes: Record<string, RouteItem<Record<string, unknown>, Component<{ routeName: string; [key: string]: unknown }>, Record<string, unknown>>>,
+  basePath: string = ''
+) {
+  const map: Record<
+    string,
+    RouteResolver<Record<string, unknown>, Component<{ routeName: string; [key: string]: unknown }>, Record<string, unknown>>
+  > = {};
 
   for (const routeName in routes) {
     const route = routes[routeName];
 
     if ('resolver' in route) {
       map[basePath + route.path] = route.resolver;
     } else if ('component' in route) {
-      const resolverClass = 'resolverClass' in route ? route.resolverClass : DefaultResolver;
+      const resolverClass = 'resolverClass' in route ? route.resolverClass! : DefaultResolver;
```

Comment:
Could we do a null coalesce here?
```suggestion
      const resolverClass = route.resolverClass ?? DefaultResolver;
```